### PR TITLE
key: 2.6.3 -> 2.10.0

### PR DESCRIPTION
**Pull request by FliegendeWurst**
###### Motivation for this change
Update to the latest version.

###### Things done
Tested that proving a simple statement works.
- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fgaz — CHANGES_REQUESTED** [review]
Thank you for doing this! I had given up on updating this package to the new gradle build...

Since the darwin failure is new, could you try to fix it or report it upstream and/or add `meta.broken = stdenv.isDarwin`?

**Comment by FliegendeWurst** [comment]
> Thank you for doing this! I had given up on updating this package to the new gradle build...
Yeah, I only managed to do it because there were already a few gradle-built packages in nixpkgs..

> Since the darwin failure is new, could you try to fix it or report it upstream and/or add meta.broken = stdenv.isDarwin?
Another option would be to skip the 2 failing tests (they don't appear to be critical to the core functionality). I'll report the issue to the mailing list too.

**Comment by fgaz** [comment]
Looks like there's an hash mismatch

**Comment by FliegendeWurst** [comment]
Hmm, now that I disabled *some* of the tests, others (that were previously passing) started to fail.. I think it will be easier to skip all of them on darwin, until the two broken tests are fixed.

File: pkgs/applications/science/logic/key/default.nix [hunk 1]
```diff
@@ -1,52 +1,86 @@
...
+    name = "KeY";
+    exec = executable-name;
+    icon = "key";
+    comment = meta.description;
+    desktopName = "KeY";
+    genericName = "KeY";
+    categories = "Science;";
+  };
+
+  gradleAction = if stdenv.isDarwin then "assemble" else "build";
```
Comment (fgaz) [inline pkgs/applications/science/logic/key/default.nix:63]: could you add a comment with a link to the upstream ticket?

Comment (FliegendeWurst) [reply pkgs/applications/science/logic/key/default.nix:63]: There is no ticket, I only sent an email. I didn't find a public bug tracker. I'll add the comment again, it was removed in one of the rebases..

**Comment by FliegendeWurst** [comment]
Another attempt: one of the developers told me that the user's home directory has to be writable for the tests to work..

**Review by fgaz — APPROVED** [review]
Looks good, thanks again!

File: pkgs/applications/science/logic/key/default.nix [hunk 2]
```diff
@@ -1,52 +1,87 @@
...
   installPhase = ''
     mkdir -p $out/share/java
-    # Wrong version in the code. On next version change 2.5 to ${version}:
-    unzip deployment/key-2.5_${gitRevision}.zip -d $out/share/java
+    cp key.ui/build/libs/key-*-exe.jar $out/share/java/KeY.jar
     mkdir -p $out/bin
+    mkdir -p $out/share/applications
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+    cp key.ui/src/main/resources/de/uka/ilkd/key/gui/images/key-color-icon-square.png $out/share/icons/hicolor/256x256/apps/key.png
+    ln -s ${desktopItem}/share/applications/* $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/key/default.nix:84]: Please use copyDesktopItems instead. Then we also do not need to create `$out/share/applications`.

**Review by fgaz — APPROVED** [review]

**Pull request merged**